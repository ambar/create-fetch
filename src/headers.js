import defaults from './defaults'

export default (headers) => defaults({headers})
