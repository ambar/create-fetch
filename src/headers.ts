import defaults from './defaults'

export default (headers: HeadersInit) => defaults({headers})
