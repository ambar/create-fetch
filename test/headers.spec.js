import headers from '../src/headers'

describe('headers', () => {
  const fetch = (...args) => Promise.resolve(args)

  it('should add headers', async () => {
    const myFetch = headers({'x-version': '1.0'})(fetch)
    expect(await myFetch('/')).toMatchSnapshot()
    expect(
      await myFetch('/', {headers: {'x-version': '1.1'}})
    ).toMatchSnapshot()
  })
})
