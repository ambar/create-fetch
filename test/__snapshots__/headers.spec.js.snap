// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`headers should add headers 1`] = `
Array [
  "/",
  Object {
    "headers": Object {
      "x-version": "1.0",
    },
  },
]
`;

exports[`headers should add headers 2`] = `
Array [
  "/",
  Object {
    "headers": Object {
      "x-version": "1.1",
    },
  },
]
`;
